stripe/src/main/java/com/stripe/android/StripePaymentController.kt

```diff
@@ -468,6 +522,53 @@ internal class StripePaymentController internal constructor(
         }
     }
 
+    /**
+     * Get the Source's client_secret from {@param data} and use to retrieve
+     * the Source object with updated status.
+     *
+     * @param data the result Intent
+     * @return the [Source] object
+     *
+     * @throws AuthenticationException failure to properly authenticate yourself (check your key)
+     * @throws InvalidRequestException your request has invalid parameters
+     * @throws APIConnectionException failure to connect to Stripe's API
+     * @throws APIException any other type of problem (for instance, a temporary issue with Stripe's servers)
+     * @throws IllegalArgumentException if the Source response's JsonParser returns null
+     */
+    @Throws(
+        AuthenticationException::class,
+        InvalidRequestException::class,
+        APIConnectionException::class,
+        APIException::class,
+        IllegalArgumentException::class
+    )
+    override suspend fun getSource(data: Intent): Source {
+        val result = PaymentFlowResult.Unvalidated.fromIntent(data)
+        val sourceId = result.sourceId.orEmpty()
+        val clientSecret = result.clientSecret.orEmpty()
+
+        val requestOptions = ApiRequest.Options(
+            apiKey = publishableKey,
+            stripeAccount = result.stripeAccountId
+        )
+
+        analyticsRequestExecutor.executeAsync(
+            analyticsRequestFactory.create(
+                analyticsDataFactory.createAuthSourceParams(
+                    AnalyticsEvent.AuthSourceResult,
+                    sourceId
+                )
+            )
+        )
```

Comment:
This logic is almost identical to `handleSourceResult`. Can `handleSourceResult` invoke `getSource`? Or maybe even better, let `Stripe` handle invoking the callback object.

Same goes for the PaymentIntent and SetupIntent versions.

Reply:
nice catch, cleaned up `PaymentController` and pushed all callback to existing logic in `Stripe.kt`, looking neat!

Reply:
Tests added to `StripePaymentAuthTest.kt`